test/libsolidity/util/TestFileParser.cpp

```diff
@@ -0,0 +1,392 @@
+		SOLT_TOKEN_LIST(T, T)
+#undef T
+		default: // Token::NUM_TOKENS:
+			return "";
+	}
+}
+
+bool TestFileParser::accept(SoltToken _token, bool const _expect)
+{
+	if (m_scanner.currentToken() == _token)
+	{
+		if (_expect)
+			expect(_token);
+		return true;
+	}
+	return false;
+}
+
+bool TestFileParser::expect(SoltToken _token, bool const _advance)
+{
+	if (m_scanner.currentToken() != _token)
+		throw Error(
+			Error::Type::ParserError,
+			"Unexpected " + formatToken(m_scanner.currentToken()) + ": \"" +
+			m_scanner.currentLiteral() + "\". " +
+			"Expected \"" + formatToken(_token) + "\"."
+			);
+	if (_advance)
+		m_scanner.scanNextToken();
+	return true;
+}
+
+string TestFileParser::parseFunctionSignature()
+{
+	string signature = m_scanner.currentLiteral();
+	expect(SoltToken::Identifier);
+
+	signature += formatToken(SoltToken::LParen);
+	expect(SoltToken::LParen);
+
+	while (!accept(SoltToken::RParen))
+	{
+		signature += m_scanner.currentLiteral();
+		expect(SoltToken::UInt);
+		while (accept(SoltToken::Comma))
+		{
+			signature += m_scanner.currentLiteral();
+			expect(SoltToken::Comma);
+			signature += m_scanner.currentLiteral();
+			expect(SoltToken::UInt);
+		}
+	}
+	signature += formatToken(SoltToken::RParen);
+	expect(SoltToken::RParen);
+	return signature;
+}
+
+u256 TestFileParser::parseFunctionCallValue()
+{
+	u256 value = convertNumber(parseNumber());
+	expect(SoltToken::Ether);
+	return value;
+}
+
+FunctionCallArgs TestFileParser::parseFunctionCallArguments()
+{
+	FunctionCallArgs arguments;
+
+	auto param = parseParameter();
+	if (param.abiType.type == ABIType::None)
+		throw Error(Error::Type::ParserError, "No argument provided.");
+	arguments.parameters.emplace_back(param);
+
+	while (accept(SoltToken::Comma, true))
+		arguments.parameters.emplace_back(parseParameter());
+	return arguments;
+}
+
+FunctionCallExpectations TestFileParser::parseFunctionCallExpectations()
+{
+	FunctionCallExpectations expectations;
+
+	auto param = parseParameter();
+	if (param.abiType.type == ABIType::None)
+	{
+		expectations.failure = false;
+		return expectations;
+	}
+	expectations.parameters.emplace_back(param);
+
+	while (accept(SoltToken::Comma, true))
+		expectations.parameters.emplace_back(parseParameter());
+
+	/// We have always one virtual parameter in the parameter list.
+	/// If its type is FAILURE, the expected result is also a REVERT etc.
+	if (expectations.parameters.at(0).abiType.type != ABIType::Failure)
+		expectations.failure = false;
+	return expectations;
+}
+
+Parameter TestFileParser::parseParameter()
+{
+	Parameter parameter;
+	if (accept(SoltToken::Newline, true))
+		parameter.format.newline = true;
+	auto literal = parseABITypeLiteral();
+	parameter.rawBytes = literal.first;
+	parameter.abiType = literal.second;
+	return parameter;
+}
+
+pair<bytes, ABIType> TestFileParser::parseABITypeLiteral()
+{
+	try
+	{
+		u256 number;
+		ABIType abiType;
```

Comment:
This should be initialized to ``ABIType::None``, shouldn't it?

Reply:
You had that other comment about the default constructor of `ABIType`. It's type is set to `None` by default. Does this need to be more explicit here? (I removed the overhead in `ABIType`).

Reply:
Let's be explicit. Added :)